### PR TITLE
спринт 10 ver. 0.1.1

### PR DESCRIPTION
**Pull request by Skzo82**

### CODE REVIEW

File: src/test/java/ru/yandex/practicum/filmorate/controller/FilmControllerTest.java
Comment (Ivanov-s-tmn) [inline src/test/java/ru/yandex/practicum/filmorate/controller/FilmControllerTest.java:1]: В тестах нужно проверить валидацию, а именно негативные сценарии (используя граничные значения)

(In the tests, you need to check the validation, namely negative scenarios (using boundary values).)

File: src/test/java/ru/yandex/practicum/filmorate/service/UserServiceTest.java
Comment (Ivanov-s-tmn) [inline src/test/java/ru/yandex/practicum/filmorate/service/UserServiceTest.java:1]: Нужны только тесты контроллеров и валидации

(Only controller tests and validations are needed)

File: src/main/java/ru/yandex/practicum/filmorate/exception/GlobalExceptionHandler.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    @ExceptionHandler(ValidationException.class)
+    public Map<String, String> handleValidationException(ValidationException ex) {
+        return Map.of("error", ex.getMessage());
+    }
+
+    @ResponseStatus(HttpStatus.NOT_FOUND)
+    @ExceptionHandler(NotFoundException.class)
+    public Map<String, String> handleNotFoundException(NotFoundException ex) {
+        return Map.of("error", ex.getMessage());
+    }
```
Comment (Ivanov-s-tmn) [inline src/main/java/ru/yandex/practicum/filmorate/exception/GlobalExceptionHandler.java:23]: Для ответов из обработчиков лучше создать отдельную дтошку по типу `ErrorResponse`, которая будет возвращаться в случае срабатывания обработки исключений

(For responses from handlers, it is better to create a separate class for example `ErrorResponse` type, which will be returned if exception handling is triggered.)

File: src/main/java/ru/yandex/practicum/filmorate/controller/FilmController.java [hunk 2]
```diff
@@ -3,64 +3,47 @@
...
     @PostMapping
     public Film createFilm(@Valid @RequestBody Film film) {
+        log.info("Добавление фильма: {}", film);
         return filmService.createFilm(film);
     }
 
-    // Обновление фильма
+    // Обновление фильма — id обязателен, остальные поля опциональны
     @PutMapping
     public Film updateFilm(@Valid @RequestBody Film film) {
```
Comment (Ivanov-s-tmn) [inline src/main/java/ru/yandex/practicum/filmorate/controller/FilmController.java:33]: Тогда нужно убрать `@Valid`, чтобы валидация не срабатывала при обновлении

(Then you need to remove `@Valid` so that validation does not trigger on update)

File: src/main/java/ru/yandex/practicum/filmorate/service/FilmService.java [hunk 3]
```diff
@@ -2,45 +2,62 @@
...
+            existingFilm.setReleaseDate(updatedFilm.getReleaseDate());
+        }
+
+        if (updatedFilm.getDuration() > 0) {
+            existingFilm.setDuration(updatedFilm.getDuration());
+        }
 
-    public List<Film> getPopularFilms(int count) {
-        return filmStorage.getPopularFilms(count);
+        return filmStorage.updateFilm(existingFilm);
```
Comment (Ivanov-s-tmn) [inline src/main/java/ru/yandex/practicum/filmorate/service/FilmService.java:61]: При обновлении пользователя логика валидации аналогичная

(When updating a user, the validation logic is similar)

File: src/main/java/ru/yandex/practicum/filmorate/storage/InMemoryUserStorage.java [hunk 4]
```diff
@@ -41,7 +41,7 @@ public List<User> findAll() {
     public User findById(int id) {
         User user = users.get(id);
         if (user == null) {
-            throw new ValidationException("Пользователь не найден");
+            throw new NotFoundException("Пользователь не найден");
         }
         return user;
```
Comment (Ivanov-s-tmn) [inline src/main/java/ru/yandex/practicum/filmorate/storage/InMemoryUserStorage.java:46]: Логично реализовать метод по подобию метода `findById` в `InMemoryFilmStorage`

(It makes sense to implement the method similar to the `findById` method in `InMemoryFilmStorage`)
